### PR TITLE
Added interval feature

### PR DESCRIPTION
**Pull request by dominno**

### CODE REVIEW

File: src/services/graphqlsettings/datasources/event.ts [hunk 1]
```diff
@@ -1,18 +1,60 @@
...
+      }
+
+    }
+
+    return dates_list;
+
+  }
+
+  async createEvent(date_created: Date, data:
+    { total_debt: string, total_balance: string, total_value_of_nfts: string, total_supply: string, number_of_loans: string, whitelisted_loans: string, ongoing_loans: string, repaid_loans: string }) {
```
Comment (ilinzweilin) [inline src/services/graphqlsettings/datasources/event.ts]: would create an interface for events and use it as a type for data

File: src/services/graphqlsettings/resolvers.ts [hunk 2]
```diff
@@ -3,14 +3,14 @@ import { eventAPI } from '../../config';
...
-      await eventAPI.findByPeriod('90d'),
+    last1d: async (_, {interval}) =>
+      await eventAPI.findByPeriod('24h', interval),
+    last7d: async (_, {interval}) =>
+      await eventAPI.findByPeriod('7d', interval),
+    last30d: async (_, {interval}) =>
+      await eventAPI.findByPeriod('30d', interval),
+    last90d: async (_, {interval}) =>
+      await eventAPI.findByPeriod('90d', interval),
   },
```
Comment (ilinzweilin) [inline src/services/graphqlsettings/resolvers.ts]: was it part of the requirements to just being able to query by these periods? And not being able to pass a custom number of days?

Comment (dominno) [reply src/services/graphqlsettings/resolvers.ts]: yes that was in requirements

File: src/services/graphqlsettings/datasources/event.ts (hunk 1 above)
Comment (dominno) [reply src/services/graphqlsettings/datasources/event.ts]: Today is my last day, so think I manage to do this, I'm very new to typescript and nodejs

**Comment by dominno** [comment]
@ilinzweilin I pushe fixes for code formating

File: src/services/graphqlsettings/datasources/event.ts (hunk 1 above)
Comment (ilinzweilin) [reply src/services/graphqlsettings/datasources/event.ts]: no worries, will take care of it

**Pull request merged**